homeassistant/components/eheimdigital/light.py

```diff
@@ -0,0 +1,157 @@
+"""EHEIM Digital lights."""
+
+from typing import Any
+
+from eheimdigital.classic_led_ctrl import EheimDigitalClassicLEDControl
+from eheimdigital.types import LightMode
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_EFFECT,
+    EFFECT_OFF,
+    ColorMode,
+    LightEntity,
+    LightEntityFeature,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.color import brightness_to_value, value_to_brightness
+
+from . import EheimDigitalConfigEntry
+from .const import EFFECT_DAYCL_MODE, EFFECT_TO_LIGHT_MODE
+from .coordinator import EheimDigitalUpdateCoordinator
+from .entity import EheimDigitalEntity
+
+BRIGHTNESS_SCALE = (1, 100)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: EheimDigitalConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up EHEIM Digital lights based on a config entry."""
+    coordinator = entry.runtime_data
+
+    known_devices: set[str] = set()
+
+    def check_devices() -> None:
+        if len(coordinator.hub.devices) == 0:
+            return
+
+        registry = er.async_get(hass)
+        existing_entities = er.async_entries_for_config_entry(registry, entry.entry_id)
```

Comment:
Why don't we get the list from the coordinator?

Reply:
How can I get the entities from the coordinator?

Reply:
You store all the identifiers as keys in your data dict, `coordinator.data.keys()`